src/database/migration/v2.js

```diff
@@ -133,8 +136,80 @@ const v2Migrations = [
     },
   },
   {
-    version: '2.3.6',
+    version: '2.3.6-IC1',
     migrate: database => {
+      // *** PART ONE, find foreign item line and delete
+
+      const itemBatches = this.database.objects('ItemBatch').snapshot();
+      const itemBatcheIDsToDelete = [];
+      // Assuming this exists everywhere
+      const inventoryAdjustmentName = database
+        .objects('Name')
+        .filtered('type = $0', 'inventory_adjustment')[0];
+
+      itemBatches.forEach(({ id, transactionBatches }) => {
+        // No transaction batches on item line should mean it's foreign
+        if (transactionBatches.length === 0) {
+          itemBatcheIDsToDelete.push(id);
+          return;
+        }
+        // supplier_invoice type transactions are all addition transactions
+        const additionTransactionBatches = transactionBatches.filtered(
+          'transaction.type = "supplier_invoice"'
+        );
+        // First try to find supplier invoice, if found this must be a local ItemBatch
+        const supplierInvoiceBatches = additionTransactionBatches.filtered(
```

Comment:
This is just an extra check? If there is a supplier invoice for the batch it is local? Or i'm missing something?

Reply:
Yeah, if there is supplier invoice, must be local item batch, since it was created via supplier invoice